module/canvas/token-placement.mjs

```diff
@@ -104,14 +107,19 @@ export default class TokenPlacement {
     this.#createPreviews();
     try {
       const placements = [];
+      let total = 0;
+      const uniqueTokens = new Map();
       while ( this.#currentPlacement < this.config.tokens.length - 1 ) {
         this.#currentPlacement++;
         const obj = canvas.tokens.preview.addChild(this.#previews[this.#currentPlacement].object);
         await obj.draw();
         obj.eventMode = "none";
         const placement = await this.#requestPlacement();
-        if ( placement ) placements.push(placement);
-        else obj.clear();
+        if ( placement ) {
+          uniqueTokens.set(placement.prototypeToken, (uniqueTokens.get(placement.prototypeToken) ?? -1) + 1);
+          placement.index = { total: total++, unique: uniqueTokens.get(placement.prototypeToken) };
```

Comment:
Is it important that we specifically track the number of unique tokens? Can we just keep a simple counter and add it to the number generated by core?

Reply:
OK, I wasn't sure how core was doing it, if it was counting unique tokens or not. It looks like it's checking based on the `actorId`, so do we want to maybe use `actorId` as the key to the `Map` rather than the `PrototypeToken` instance?